### ISSUE TITLE
feature request: passing File::open value to where Write is expected

### ISSUE THREAD

**Issue by KisaragiEffective** [rust-lang/rust-clippy#8608]
### What it does

detects passing `File::open` value to place where `Write` is expected

### Lint Name

file_is_not_writable

### Category

suspicious

### Advantage

- Remove "bad file descriptor" errors at runtime.
 

### Drawbacks

none

### Example

```rust
let file = File::open("config.txt"); // this handle is read-only
let buf_write = BufWriter::new(file);
```

Could be written as below when above 1.58.0:

```rust
let file = File::options().write(true).open("config.txt");
let buf_write = BufWriter::new(file);
```

when below 1.58.0:
```rust
let file = OpenOptions::new().write(true).open("config.txt");
let buf_write = BufWriter::new(file);
```
